src/views/blocks/rsvp.php

```diff
@@ -13,27 +13,52 @@
  *
  */
 
-$event_id = $this->get( 'post_id' );
-$tickets  = $this->get( 'tickets' );
+$event_id    = $this->get( 'post_id' );
+$tickets     = $this->get( 'tickets' );
+$num_tickets = count( $tickets );
 
-?>
-
-<?php $this->template( 'blocks/attendees/order-links', array( 'type' => 'RSVP' ) ); ?>
+// We don't display anything if there is no RSVP
+if ( empty( $num_tickets ) ) {
+	return false;
+}
 
-<div class="tribe-block tribe-block__rsvp">
+// Get active RSVPs
+$active_tickets = array();
 
-	<?php foreach ( $tickets as $ticket ) : ?>
+foreach ( $tickets as $ticket ) {
+	if ( tribe_events_ticket_is_on_sale( $ticket ) ) {
+		$active_tickets[] = $ticket;
+	}
+}
 
-		<div class="tribe-block__rsvp__ticket" data-rsvp-id="<?php echo absint( $ticket->ID ); ?>">
+$num_active_tickets = count( $active_tickets );
+$has_active_tickets = ! empty( $num_active_tickets );
 
-			<?php $this->template( 'blocks/rsvp/icon' ); ?>
+if ( ! $has_active_tickets ) {
+	$active_past = true;
+	$timestamp   = current_time( 'timestamp' );
 
-			<?php $this->template( 'blocks/rsvp/content', array( 'ticket' => $ticket ) ); ?>
+	foreach ( $tickets as $ticket ) {
+		$active_past = ( $active_past && $ticket->date_is_later( $timestamp ) );
+	}
+}
+?>
 
-			<?php $this->template( 'blocks/rsvp/loader' ); ?>
+<?php $this->template( 'blocks/attendees/order-links', array( 'type' => 'RSVP' ) ); ?>
 
+<div class="tribe-block tribe-block__rsvp">
+	<?php if ( $has_active_tickets ) : ?>
+		<?php foreach ( $active_tickets as $ticket ) : ?>
```

Comment:
I would say, we add a marker on the ticket to say if it's active or not outside of the view and here we only display what we need based on that. 

Allowing for a simpler view.